src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -162,9 +170,10 @@ public void Dispose()
                             ErrorHandler.ThrowOnFailure(service.UnadviseDirChange(_directoryWatchCookie));
                         }
 
+                        // shouldn't _activeFileWatchingTokens need a lock or need local copy before the queue?
```

Comment:
my worry was more on data corruption where other thread modified the HashSet but since Dispose doesn't have memory barrier, not all that changes are synched to the thread Disposed is called?